### PR TITLE
Fix debugger breakpoints

### PR DESCRIPTION
**Pull request by elad335**
* Fix an old regression from #7905 which made breakpoints literally not appear after setting them, they would only appear after scrolling which triggers the missing call to `ShowAddress`. Because this function is also responsible for refreshing breakpoints viewing list.
* Fix foreground/background color of text when encountering non-execuable memory, this bug always existed and can be reproduced by simply jumping to address 0 after setting a breakpoint. you would see that the breakpoints colors are still placed.

### CODE REVIEW

**Comment by Megamouse** [comment]
This is not entirely correct.
The proper way to fix this is to call 
`Q_EMIT RequestShowAddress(pc);`
in AddBreakPoint

**Comment by elad335** [comment]
Fixed.

**Comment by elad335** [comment]
Added another fix.

File: rpcs3/rpcs3qt/debugger_list.cpp [hunk 1]
```diff
@@ -99,6 +101,8 @@ void debugger_list::ShowAddress(u32 addr, bool force)
 			if (!vm::check_addr(cpu_offset + pc, 4))
 			{
 				item(i)->setText((IsBreakpoint(pc) ? ">> " : "   ") + qstr(fmt::format("[%08x]  ?? ?? ?? ??:", pc)));
+				item(i)->setForeground(palette().color(foregroundRole()));
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/debugger_list.cpp:104]: wouldn't it make more sense to simply move the entire section that sets these to the beginning of the loop scope ?

Comment (elad335) [reply rpcs3/rpcs3qt/debugger_list.cpp]: Review the latest changes please.

Comment (Megamouse) [reply rpcs3/rpcs3qt/debugger_list.cpp]: when will you stop producing the most unreadable code in history

Comment (elad335) [reply rpcs3/rpcs3qt/debugger_list.cpp]: Okay for now.

**Pull request merged**